hazelcast/src/test/java/com/hazelcast/map/impl/nearcache/NearCacheTestSupport.java

```diff
@@ -86,34 +88,48 @@ protected void testNearCacheEviction(IMap<Integer, Integer> map, int size) {
         assertEquals("got the wrong eviction count", expectedEvictions, stats.getEvictions());
         assertEquals("got the wrong expiration count", 0, stats.getExpirations());
         assertEquals("we expect the same hits", statsBeforeEviction.getHits(), stats.getHits());
-        assertEquals("we expect the same misses", statsBeforeEviction.getMisses(), stats.getMisses());
+        assertEquals("we expect one miss more", statsBeforeEviction.getMisses() + 1, stats.getMisses());
     }
 
     protected void testNearCacheExpiration(final IMap<Integer, Integer> map, final int size, int expireSeconds) {
         populateMap(map, size);
         populateNearCache(map, size);
 
-        final NearCacheStats statsBeforeExpiration = getNearCacheStats(map);
-        assertTrue(format("we expected to have all map entries in the Near Cache or already expired (%s)", statsBeforeExpiration),
-                statsBeforeExpiration.getOwnedEntryCount() + statsBeforeExpiration.getExpirations() >= size);
+        final NearCacheStats statsBeforeExpiration = getNearCacheStatsCopy(map);
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() throws Exception {
+                assertTrue(format("we expected to have all map entries in the Near Cache or already expired (%s)",
+                        statsBeforeExpiration),
+                        statsBeforeExpiration.getOwnedEntryCount() + statsBeforeExpiration.getExpirations() >= size);
+            }
+        });
 
         sleepSeconds(expireSeconds + 1);
 
+        final AtomicInteger invocationCounter = new AtomicInteger();
         assertTrueEventually(new AssertTask() {
             public void run() {
                 // map.get() triggers Near Cache eviction/expiration process,
                 // but we need to call this on every assert since the Near Cache has a cooldown for expiration cleanups
                 map.get(0);
+                long invocations = invocationCounter.incrementAndGet();
 
-                NearCacheStats stats = getNearCacheStats(map);
-                assertEquals("we expect the same hits", statsBeforeExpiration.getHits(), stats.getHits());
-                assertEquals("we expect the same misses", statsBeforeExpiration.getMisses(), stats.getMisses());
-                assertEquals("we expect all entries beside the 'trigger entry' to be deleted from the Near Cache",
+                NearCacheStats stats = getNearCacheStatsCopy(map);
```

Comment:
Here I get a stats copy again. The copy may suffer from the non-atomic updates, but at least the numbers don't change during the asserts. Since I print the whole stats in the assert message it could be very confusing otherwise, if one gets something like `we expect 5000 expirations, but was 5 (NearCachStatsImpl{..., expiration=6, ...})` due to ongoing evictions.